Add basic version of CoC based on the last one

escherlies opened the pull request:
To align this more with the current design, i created a basic version of the CoC.

I think we should publish this asap and refine/redo the contents of the CoC later.

Todo:
- [x] Add version
- [x] Remove .md
- [x] Should we place this on the Main Menu or is it OK to have on the Landing Page like it is now?

escherlies committed:
Add COC from 2024 as base

escherlies committed:
Restructure Code of Conduct

- Reorganize content into clearly defined sections and principles
- Group related principles under core themes for better readability
- Add a dedicated section for reporting and assistance information
- Update title for NixCon 2025 and indicate a minimal base version

escherlies committed:
Add Code of Conduct page

- Remove the downloadable PDF version of the Code of Conduct
- Create a new web page at `/coc` to display the Code of Conduct directly
- Update navigation links on mobile and desktop menus
- Update the home page link to view the new Code of Conduct page

escherlies committed:
Remove placeholder text from Code of Conduct

escherlies committed:
Clean up bullet point formatting

escherlies commented:
Related #17

das-g commented:
If the content of the CoC is to be published, yet still subject to change, should we date or version it and/or explicitly refer to the website Git history for those interested in how it (will have) changed over time?

das-g commented on COC.md:
```diff
@@ -0,0 +1,35 @@
+**NixCon 2025 - Code of Conduct**
```
Is there a reason to use strong emphasis rather than semantic title markup?

das-g commented:
How are `COC.md` and `src/routes/coc.jsx` to be synched with each other? Is there a static site generator in place or is that a manual process? If the latter, what's the purpose of  `COC.md`? Ease of content editing and reviewing for non-webdevs?

infinisil commented on COC.md:
We probably don't need to duplicate this here, we can link people to the website version imo.

escherlies commented:
> If the content of the CoC is to be published, yet still subject to change, should we date or version it and/or explicitly refer to the website Git history for those interested in how it (will have) changed over time?

Sounds good. Let's add a timestamp to it. 

No need for explicit refering i think, since I want to place a link to the repo anyway somewhere on the site

escherlies commented on src/components/MobileMenu.jsx:
```diff
@@ -101,6 +102,20 @@ export default function MobileMenu() {
               Organizers
             </A>
+
+            {/* Code of Conduct */}
```
Do we want the `CoC` in the main menu or is it enough to have it on the landing page?

das-g replied:
How about in the footer?

> [animation toggle] | [Legal disclosure](https://2025.nixcon.org/legal) | [Privacy Policy](https://2025.nixcon.org/privacy) | <ins>[Code of Conduct](https://2025.nixcon.org/coc)</ins>

That way it's accessible from all pages of the site without taking up precious top menu space. (It's probably worth though to *additionally* keep it linked from a prominent position such as the landing page main content, so that also people who aren't explicitly looking for it will be aware that there's a CoC. Even better would be to make people aware of this fact (also?) in the ticket reservation process.)

escherlies replied:
Sounds good!

escherlies committed:
Remove redundand COC.md

escherlies committed:
Add last updated date to Code of Conduct

das-g commented on src/routes/coc.jsx:
```diff
@@ -0,0 +1,87 @@
+
+          <div class={cubContainerClasses}>
+            <p class="font-semibold text-lg mb-2">NixCon 2025 - Code of Conduct</p>
+            <p class="text-sm mb-2">Last updated: May 29, 2025</p>
```
```suggestion
            <p class="font-semibold text-lg mb-2">NixCon 2025 - Code of Conduct</p>
            <p>This Code of Conduct is still subject to change. Check back here for the latest version.</p>
            <p class="text-sm mb-2">Last updated: May 29, 2025</p>
```

escherlies committed:
Update src/routes/coc.jsx

Co-authored-by: Raphael Das Gupta <github.com@raphael.dasgupta.ch>

escherlies committed:
Move Code of Conduct link to footer

- Relocate Code of Conduct link to the site footer
- Remove Code of Conduct link from top and mobile navigation menus
